src/reaktoro_pse/reaktoro_block.py

```diff
@@ -0,0 +1,952 @@
+        "max_iters",
+        ConfigValue(
+            default=400,
+            domain=int,
+            description="Maximum number of iterations for reaktoro solver",
+            doc="""The maximum number of iterations for reaktoro solver""",
+        ),
+    )
+    CONFIG.declare(
+        "presolve_during_initialization",
+        ConfigValue(
+            default=False,
+            domain=bool,
+            description="Option to pre-solve to low tolerance first, before primary solve but only during initialization",
+            doc="""In some cases reaktoro might fail to solve to high tolerance first,
+            a presolve at low tolerance can enable the reaktoro solve to high tolerance, this will only presolve during initialization""",
+        ),
+    )
+    CONFIG.declare(
+        "presolve_property_block",
+        ConfigValue(
+            default=False,
+            domain=bool,
+            description="Option to pre-solve to low tolerance first on main property block, before primary solve",
+            doc="""In some cases reaktoro might fail to solve to high tolerance first,
+            a presolve at low tolerance can enable the reaktoro solve to high tolerance""",
+        ),
+    )
+    CONFIG.declare(
+        "presolve_speciation_block",
+        ConfigValue(
+            default=False,
+            domain=bool,
+            description="Option to pre-solve to low tolerance for specification block, before primary solve",
+            doc="""In some cases reaktoro might fail to solve to high tolerance first,
+            a presolve at low tolerance can enable the reaktoro solve to high tolerance""",
+        ),
+    )
+    CONFIG.declare(
+        "presolve_max_iters",
+        ConfigValue(
+            default=400,
+            domain=int,
+            description="Presolve maximum number of iterations for reaktoro solver",
+            doc="""The maximum number of iterations for reaktoro pre-solver""",
+        ),
+    )
+    CONFIG.declare(
+        "presolve_tolerance",
+        ConfigValue(
+            default=1e-4,
+            domain=float,
+            description="Tolerance for reaktoro pre-solve",
+            doc="""Tolerance for reaktoro pre-solver""",
+        ),
+    )
+    CONFIG.declare(
+        "presolve_epsilon",
+        ConfigValue(
+            default=1e-32,
+            domain=float,
+            description="Tolerance for reaktoro pre-solve",
+            doc="""Defines what is considered to be 0 for ion composition""",
+        ),
+    )
+    CONFIG.declare(
+        "hessian_type",
+        ConfigValue(
+            default="Jt.J",
+            domain=IsInstance((str, HessTypes)),
+            description="Hessian type to use for reaktor gray box",
+            doc="""Hessian type to use, some might provide better stability
+                options (Jt.J, BFGS, BFGS-mod,BFGS-damp, BFGS-ipopt""",
+        ),
+    )
+    CONFIG.declare(
+        "open_species_on_property_block",
```

Comment:
We will need some documentation on what this means.

Reply:
yeap.